subprojects/packagefiles/fmt/meson.build

```diff
@@ -15,40 +15,34 @@ endif
 
 header_only = get_option('header-only')
 
-fmt_lib = library(
-  'fmt',
-  'src/format.cc',
-  'src/os.cc',
-  cpp_args: fmt_private_cpp_args,
-  include_directories: 'include',
-  build_by_default: not header_only,
-  install: not header_only,
-)
-
-if not header_only
+if header_only
+  fmt_header_only_dep = declare_dependency(
+    include_directories: 'include',
+    compile_args: '-DFMT_HEADER_ONLY',
+  )
+else
+  fmt_lib = library(
+    'fmt',
+    'src/format.cc',
+    'src/os.cc',
+    cpp_args: fmt_private_cpp_args,
+    include_directories: 'include',
+    build_by_default: not header_only,
```

Comment:
This and below should be possible to set to `true` as we know that header_only is false here.